iceoryx_hoofs/source/posix_wrapper/shared_memory_object/shared_memory.cpp

```diff
@@ -176,19 +197,22 @@ bool SharedMemory::open(const int oflags, const mode_t permissions, const uint64
     return true;
 }
 
+bool SharedMemory::unlinkIfExist(const Name_t& name) noexcept
+{
+    return !posixCall(shm_unlink)(name.c_str())
+                .failureReturnValue(INVALID_HANDLE)
+                .suppressErrorMessagesForErrnos(ENOENT)
```

Comment:
Shouldn't this be `.ignoreErrnos(ENOENT)`? After all the method has an `IfExists` suffix

Reply:
No the `IpcChannel` contract is to return true if the channel was removed otherwise false.

Reply:
That's true for `unlink` but this is called `unlikIfExists` which, for me, implies a different semantic like "I only want to handle a failure if it is still there after this call but don't care if it wasn't there in the first place". Maybe instead of a binary return value, an expected would be necessary?

Oh, and please always add doxygen documentation for new code. You don't need to add it to all the methods which are already there, since this is out of scope for the PR but the new code should have a documentation.

Reply:
Okay, now I understand. This class wasn't used for an `IpcChannel` implementation and therefore the method is new. The `IpcChannel` implementations have a `cxx::expected<bool, Error> unlinkIfExists(const Name_t& name)` signature though.

Reply:
returning an expected now